recipes/pypy3.6/meta.yaml

```diff
@@ -0,0 +1,81 @@
+{% set name = "pypy3.6" %}
+{% set version = "7.1.0b" %}
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  - url: https://bitbucket.org/pypy/pypy/downloads/pypy3.6-v7.1.0-src.tar.bz2
+    fn: pypy3.6-v7.1.0-src.tar.bz2
+    sha256: faa81f469bb2a7cbd22c64f22d4b4ddc5a1f7c798d43b7919b629b932f9b1c6f
+    folder: pypy3
+    patches:
+      - fficurses.patch
+      - minimal_curses.patch
+      - tklib_build.patch
+      - clibffi.patch  # [osx]
+      - darwin.patch  # [osx]
+
+  - url: https://bitbucket.org/pypy/pypy/downloads/pypy2-v5.9.0-osx64.tar.bz2  # [osx]
+    fn: pypy2-v5.9.0-osx64.tar.bz2  # [osx]
+    sha256: 94de50ed80c7f6392ed356c03fd54cdc84858df43ad21e9e971d1b6da0f6b867  # [osx]
+    folder: pypy2-osx  # [osx]
+
+build:
+  number: 0
+  skip: True  # [win]
+  skip_compile_pyc:
+    - lib*
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - python >=2.7,<3
+  host:
+    - python >=2.7,<3
+    - libunwind >=4.0  # [osx]
+    - pkg-config
+    - pycparser
+    - openssl
+    - libffi
+    - sqlite
+    - tk
+    - zlib
+    - bzip2
+    - expat
+    - ncurses >=6.0
+    - gdbm
+    - xz
+    - tk
+    - bzip2
+    - xorg-libx11  # [linux]
+  run:
+    - libunwind >=4.0  # [osx]
+    - openssl
+    - libffi
+    - sqlite
+    - tk
+    - zlib
+    - bzip2
+    - ncurses >=6.0
+    - gdbm
+    - xz
+    - expat
+    - bzip2
+
+test:
+  commands:
+    - pypy3 --help
```

Comment:
I'm not vary familiar with PyPy but should this conflict with `pypy3.5`? Or does it make more sense for them as two branches in a single feedstock called `pypy3`?

Reply:
PyPy release a separate version for each interpreter version they support, so they have:
PyPy2.7 (version 7.1.0) - Python 2.7
PyPy3.5 (version 7.0.0) - Python 3.5 syntax + f-strings
PyPy3.6 (version 7.1.0) - Python 3.6 syntax

So all of these exists at the same time, and when we originally added the `pypy3.5` feedstock we wanted to keep it as close to that release naming as possible.

For example right now PyPy3.6 is a *beta*, so for production 3.5 is better, but they release minors for 3.6 (so it is updated separately).

The resulting executable is called `pypy3` BTW.

We could have a single `pypy3` feedstock, which might also be easier for users to discover :)

Reply:
Right. The issue here is if a user has both pyp3.5 and 3.6 installed, then the single `pypy3` executable will be stomped on by one package or the other, right?

Reply:
Yes, but it's also possible that a user will want to specify a version, so `pypy3==3.5.3-7.0.0` is not very ergonomic.

Also because running pypy requires a new empty env, this is not likely to happen to users.

Reply:
Ahhhhh I see. This is why I am not do CR here. :p

Reply:
The versioning is unfortunate. How about forcing it to conflict with `pypy3.5` then to ensure nobody ends up with a broken environment?
```yaml
requirements:
  run_constrained:
    - pypy3.5 ==99999999999
```
(It's a crude but this makes pypy3.5 an optional dependency but constrains the version to be one that will never exist, hence it is impossible to install both in the same environment.)

Reply:
Is this trick in the conda-forge docs? I was trying to find it and missed it.

Reply:
I can't find it there either

Reply:
mmk - lets ping @croth1 and get it added

Reply:
I think we want for now to just go with `pypy3.6`. 
It'll be easier for me to maintain both, and it's not really an issue for PyPy users since they won't just install two packages at the same time.
What do you say?

Reply:
I'm pretty agnostic here so whatever you think is best! :)

Reply:
The run_constrained thing is more of a safe guard than a shift in how the package works. I'd be in favor of adding that.

Reply:
Sure, where do I put it?

Reply:
In the requirements section per the comment from @chrisburr above.